src/connector.rs

```diff
@@ -198,29 +324,95 @@ impl EthConnectorContract {
         .unwrap();
         // Burn tokens to recipient
         let predecessor_account_id = String::from_utf8(sdk::predecessor_account_id()).unwrap();
-        self.burn(predecessor_account_id, args.amount);
+        self.burn_near(predecessor_account_id, args.amount);
         // Save new contract data
         self.save_contract();
         sdk::return_output(&res[..]);
     }
 
+    /// Withdraw ETH tokens
+    pub fn withdraw_eth(&mut self) {
+        use crate::prover;
+        #[cfg(feature = "log")]
+        sdk::log("Start withdraw ETH".into());
+        let args: WithdrawEthCallArgs = WithdrawEthCallArgs::from(
+            parse_json(&sdk::read_input()).expect(str_from_slice(FAILED_PARSE)),
+        );
+        assert!(
+            prover::verify_withdraw_eip712(
+                args.sender,
+                args.eth_recipient,
+                self.contract.eth_custodian_address,
+                args.amount,
+                args.eip712_signature
+            ),
+            "ERR_WRONG_EIP712_MSG"
+        );
+
+        let res = WithdrawResult {
+            recipient_id: args.eth_recipient,
+            amount: args.amount.as_u128(),
+            eth_custodian_address: self.contract.eth_custodian_address,
+        }
+        .try_to_vec()
+        .unwrap();
+        // Burn tokens to recipient
+        self.burn_eth(args.eth_recipient, args.amount.as_u128());
+        // Save new contract data
+        self.save_contract();
+        sdk::return_output(&res[..]);
+    }
+
+    // Return total supply of NEAR + ETH
     pub fn ft_total_supply(&self) {
         let total_supply = self.ft.ft_total_supply();
         sdk::return_output(&total_supply.to_string().as_bytes());
         #[cfg(feature = "log")]
         sdk::log(format!("Total supply: {}", total_supply));
     }
 
+    // Return total supply of NEAR
+    pub fn ft_total_supply_near(&self) {
+        let total_supply = self.ft.ft_total_supply_near();
+        sdk::return_output(&total_supply.to_string().as_bytes());
+        #[cfg(feature = "log")]
+        sdk::log(format!("Total supply NEAR: {}", total_supply));
```

Comment:
Seems a little counter-intuitive to put the logging _after_ the return. I would expect a `return` to be the last statement and all following code unreachable. But maybe that is not the case with this `return_output` API? I guess it just sets some state internal to the Near contract runtime which is used as the return value after the wasm execution finishes?

In any case, I think the code would map closer to normal rust patterns if `return_output` acted similar to the `return` keyword.